switchtec.c

```diff
@@ -363,7 +363,7 @@ static ssize_t field ## _show(struct device *dev, \
 	struct switchtec_dev *stdev = to_stdev(dev); \
 	struct sys_info_regs __iomem *si = stdev->mmio_sys_info; \
 	\
-	if (stdev->gen == SWITCHTEC_GEN4) \
+	if (stdev->gen == SWITCHTEC_GEN4 || stdev->gen == SWITCHTEC_GEN5) \
```

Comment:
Should we maybe just make this >= GEN4? Instead of creating a list that needs to be added to next generation?

Reply:
I second it. We can use >= GEN4 till there's incompatibility in the hardware so that we must distinguish GEN4, GEN5.

Reply:
Code updated. Please review and approve if nothing else.